### PR TITLE
Update history and version number for Taylor release v1.2.0

### PR DESCRIPTION
**Pull request by jackiekazil**
Ticket #1599

### CODE REVIEW

**Comment by jackiekazil** [comment]
@rht any insights into why I am failing on lint-ruff?

**Comment by rht** [comment]
I will fix the Ruff errors and pin the version. It's changing too fast.

**Comment by rht** [comment]
Is going to be fixed in #1609.

File: HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,85 @@
...
+
+**Special notes**
+
+New features:
+
+* Implement radius for NetworkGrid.get_neighbors #1564
+
+Some highlights for the perf improvements:
+
+* Use getattr for attribute strings in model data collection #1590 this is a 2x speedup over the relevant line, but not sure if compared to the rest of the function
```
Comment (rht) [inline HISTORY.rst:18]: We should probably either remove the "but not sure if compared to the rest of the function" part or benchmark the overall speedup.

File: HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,85 @@
...
+* Refactor iter/get_cell_list_contents methods #1570 at least 1.3x speedup for iter/get_cell_list_contents
+* Evaluate empties set more lazily #1546 (comment) ~1.3x speedup for place_agent, remove_agent, and move_agent
```
Comment (rht) [inline HISTORY.rst:20]: Needs to add 1 more:
https://github.com/projectmesa/mesa/pull/1597 is for 1.3x speedup for grid access (`grid[x, y]`).

**Comment by rht** [comment]
Reminder that the commit message needs to be in imperative form.

**Comment by jackiekazil** [comment]
@rht should all be addressed. Also added the last pr by tpike.

**Comment by rht** [comment]
LGTM

**Pull request merged**